Index: scikit-learn-intelex-2021.4.0/generator/wrapper_gen.py
===================================================================
--- scikit-learn-intelex-2021.4.0.orig/generator/wrapper_gen.py
+++ scikit-learn-intelex-2021.4.0/generator/wrapper_gen.py
@@ -1356,7 +1356,7 @@ class wrapper_gen(object):
         """
         return code for initing
         """
-        cpp = "#ifndef DAAL4PY_CPP_INC_\n" + \
+        cpp = "#include <stdlib.h>\n#ifndef DAAL4PY_CPP_INC_\n" + \
             "#define DAAL4PY_CPP_INC_\n#include <daal4py_dist.h>\n\n"
         pyx = ''
         for i in self.ifaces:
