From 0178bc28350b03658bf9e8887f257ed3a4ff2602 Mon Sep 17 00:00:00 2001
From: Paul Yim <paul.j.yim@gmail.com>
Date: Tue, 27 Sep 2022 18:22:40 -0400
Subject: [PATCH] Hardcode extension suffix; distutils seems to be broken for
 unix py<39

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 864ab5d..510b909 100755
--- a/setup.py
+++ b/setup.py
@@ -323,7 +323,7 @@ def build_oneapi_backend():
 
 
 def get_onedal_py_libs():
-    ext_suffix = get_config_vars('EXT_SUFFIX')[0]
+    ext_suffix = '.so'
     libs = [f'_onedal_py_host{ext_suffix}', f'_onedal_py_dpc{ext_suffix}']
     if IS_WIN:
         ext_suffix_lib = ext_suffix.replace('.dll', '.lib')
-- 
2.32.1 (Apple Git-133)

